src/main/java/com/ase/restservice/controller/AccountController.java

```diff
@@ -42,33 +45,33 @@ public Account createAccount(@Valid @RequestBody Account account) {
    * @throws ResourceNotFoundException if account does not exist in the database
    */
   @GetMapping("/accounts/{accountId}/balance")
-  public Account getAccountBalance(@PathVariable(value = "accountId")
-      String accountId) throws ResourceNotFoundException {
-    Account account = accountRepository.findById(accountId)
-        .orElseThrow(() -> new ResourceNotFoundException(
-            "Account not found for this id :: " + accountId
-        ));
-    return account;
+  public Float getAccountBalance(@PathVariable(value = "accountId") String accountId)
+      throws ResourceNotFoundException {
+    Account account = accountService.findById(accountId);
+    return account.getBalance();
   }
 
   /**
    * Update account balance.
    *
    * @param accountId AccountID
    * @param amount Value that will be summed with balance
-   * @return Updated account.
+   * @return Updated account
    * @throws ResourceNotFoundException if account does not exist in the database
    */
   @PutMapping("/accounts/{accountId}/balance")
-  public Account updateAccountBalance(@PathVariable(value = "accountId")
-      String accountId, @RequestParam(value = "amount", defaultValue = "0") String amount)
+  public Account updateAccountBalance(@PathVariable(value = "accountId") String accountId,
```

Comment:
The logic in this function seems like it should be in the service layer

Reply:
Changed it